webapp/lib/controller.dart

```diff
@@ -372,22 +372,26 @@ void updateNeedsReplyCharts(List<model.NeedsReplyData> filteredNeedsReplyDataLis
   Map<DateTime, int> data = new Map.fromIterable(filteredNeedsReplyDataList,
     key: (item) => (item as model.NeedsReplyData).datetime.toLocal(),
     value: (item) => (item as model.NeedsReplyData).needsReplyCount);
-  view.contentView.needsReplyTimeseries.updateChart([data], timeScaleUnit: timeScaleUnit);
+  DateTime xUpperLimitDateTime = getEndDateTimeForPeriod();
```

Comment:
Could these be done the same way you did for system events?

ie. define `xLowerLimitDateTime` and `xUpperLimitDateTime` only once at the top of this method after `timeScaleUnit`, and then just reference them in `updateChart as you're doing

I think it would be good to set both the lower and upper datetime limit always (it will make the charts align, and makes the UI more usable when switching between periods - you know for sure which period you're looking at, rather than letting the chart adjust based on the data, which may be missing)

Reply:
Yes, that a good idea. I've refactored it to be the same way